### ISSUE TITLE
Darken/lighten colours need checking

### ISSUE THREAD

**Issue by lah7** [lah7/ubuntu-mate-colours#4]
Some of the colours brightness adjustments are not quite ready yet, some of them are a bit too dark.

---

Checkboxes for **aqua** have very dark gradients:

![Screenshot_20191005_113856](https://user-images.githubusercontent.com/13032135/66253754-b8854b80-e764-11e9-85f9-612c3124b296.png)

Folder icons for **brown** and some others also are very dark:

![Screenshot_20191005_114050](https://user-images.githubusercontent.com/13032135/66253767-fc785080-e764-11e9-934d-cf902883252e.png)

It looks OK on some colours, however:

![Screenshot_20191005_114120](https://user-images.githubusercontent.com/13032135/66253768-0e59f380-e765-11e9-9e81-cf515feddb43.png)

**Comment by lah7** [comment]
I've re-tuned the gradients, but still feel like they need tweaking. Perhaps the hex colour need adjusting to a brighter, more vibrant colour.

**These icons are looking good:**

* Aqua
* Blue
* Pink
* Purple
* Red


**These still have a dark gradient on folders:**

* Brown
* Green
* Orange
* Teal
* Yellow

**Comment by cerisezero** [comment]
A possible, if more complex, solution to creating a more natural/contrasting gradient automatically might be to convert the colour values to HSL and lighten/darken them that way before converting back, using the lightness differences present in the original images as a guide? Not sure if that might just be adding too many steps, but a quick manually edited test looks promising:
![example](https://user-images.githubusercontent.com/56359596/67214046-fb0e7f80-f416-11e9-94b9-66a8a39a07ac.png)
Left is the default, middle is the current auto-generated version, right uses HSL differences to create gradient. I'll look further into it.

**Comment by lah7** [comment]
Thanks for having a look, using HSL is a good idea. :+1:

In the code, there's a common function that could be updated - `_get_hex_variant`:

https://github.com/lah7/ubuntu-mate-colours/blob/16fcda74a3777eee9673901284731b8909f1cb83/ubuntu-mate-colours-generator#L217-L228

I suppose it could pass the original green hex (`#87A752`) and the variant shade of green (e.g. `#4e6827`) to get the brightness difference, and then apply that difference to the new colour (`#E231A3` for pink)

**Comment by cerisezero** [comment]
That's more or less what I'd been thinking; I'd mostly been looking at just hardcoding in brightness differences as with the current gradients, given that the colours in the original files aren't changing, but having the program get the brightness difference automatically would be a neat way to do it.

**Comment by cerisezero** [comment]
Okay, tested it with a HSL based gradient with hardcoded brightness changes, and the results are looking good. 

![Screenshot at 2019-10-29 12-10-34](https://user-images.githubusercontent.com/56359596/67766166-b94c8d00-fa45-11e9-8765-28e47d42fed5.png)
![Screenshot at 2019-10-29 12-14-28](https://user-images.githubusercontent.com/56359596/67766167-b94c8d00-fa45-11e9-94f7-4fe6ae44d818.png)
![Screenshot at 2019-10-29 12-14-45](https://user-images.githubusercontent.com/56359596/67766168-b9e52380-fa45-11e9-8464-7604c68bc935.png)

My code's pretty rough and ready at the moment - it requires converting the hex values to RGB and then to HSL, so the current implementation is converting some colour entries twice for usage in the CSS - but it works...

**Comment by lah7** [comment]
Having two conversions is not a problem - it'll likely be a trivial amount of processing. The generation happens at build time, which already takes quite some time for 27 theme combinations.

When you're ready, please create a pull request. I'll look at some other issues and then trigger a new round of  packages. :+1:

**Comment by cerisezero** [comment]
Pull request should be waiting. Added the missing definitions from bug #15 at the same time as they were in my personal copy of the file; I hope this doesn't cause any complications.

**Comment by lah7** [comment]
Separate commits do help make things _less complicated._ :smile: I did notice your [`patch-1`](https://github.com/cerisezero/ubuntu-mate-colours/tree/patch-1) branch and was about to cherry pick that, but then realised your pull request (#16) did indeed include them.

Going to test and trigger a new round of packages, based on `ubuntu-mate-artwork` 19.10.10.

**Comment by cerisezero** [comment]
Sorry about that! I'll keep things cleaner in future.

**Comment by lah7** [comment]
No worries! I've generated a new set locally, and found a few "black spots" (let's call it) on some colours:

![image](https://user-images.githubusercontent.com/13032135/67989766-bca16d80-fc2b-11e9-83eb-ed02a4add736.png)

![image](https://user-images.githubusercontent.com/13032135/67989918-405b5a00-fc2c-11e9-8b25-f50c5227b1f7.png)

![image](https://user-images.githubusercontent.com/13032135/67990059-a647e180-fc2c-11e9-89e2-98ed749ce6fb.png)

![image](https://user-images.githubusercontent.com/13032135/67990133-de4f2480-fc2c-11e9-8b9f-1ffe7bc7f955.png)

With the help of some `print()` statements, I found out the resulting RGB / hex values were outside the 0-255 range, so this happened:

    28 -> #36d1102 | [54, 209, 258]

I added validation in d475ccec6e07edb9a6d3478670eea2bde6bd1863 which has fixed this. :paintbrush: 

We can consider the original brightness issues solved. :medal_sports:
